### ISSUE TITLE
Full support for tablets in Alternator - and make it the default

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#16355]
After https://github.com/scylladb/scylladb/issues/16203 was fixed (in PR https://github.com/scylladb/scylladb/pull/16353), we can now create an Alternator table that uses **tablets** instead of vnodes by setting the `experimental:initial_tablets` tag on the new table.

The goal of this issue is to make this behavior the default. I.e., every new Alternator table should use tablets, not vnodes.

Before we can enable tablets on Alternator _by default_, we need to make sure that 100% of the Alternator features supported with vnodes will also work correctly with tables. PR https://github.com/scylladb/scylladb/pull/16353 fixed one major broken feature (secondary indexes), but additionally the following features are thought to be broken and will need to be fixed:

1. **Alternator Streams** needs CDC, which isn't yet supported on tablets (https://github.com/scylladb/scylladb/issues/16317).
2. **Alternator TTL** doesn't work (https://github.com/scylladb/scylladb/issues/16567)

To be sure this issue is really done, we'll need to run all Alternator tests - both in-tree test/alternator and dtest's Alternator tests - using tablets and see all of them pass. Right now, all Alternator Streams and Alternator TTL tests fail when tablets are forced in Alternator.

**Comment by nyh** [comment]
I just remembered that Alternator Streams is still considered **experimental** (it was on the 2023 plan to fix it, but it was postponed indefinitely, CC @mykaul @tzach) and not enabled without a special "--experimental" option. So it's not actually critical to support it to announce that Alternator fully supports tablets.

**Comment by nyh** [comment]
Recently, Scylla began to enable tablets on CQL keyspaces by default if the "tablets" experimental feature is on, and it no longer requires an explicit "initial tablets" setting. This is despite the fact that we know that not all features are supported yet with tablets (e.g., CDC). So it's time to do the same with Alternator now. What I need to do now is:

1. Enable tablets (if the experimental feature exists) on new Alternator tables, without requiring a special experimental tag.
2. The way to enable tablets on the low-level code got changed and needs to be fixed (I suspect the current code doesn't even work!):  `REPLICATION = {... 'initial_tablets': ...}` to `TABLETS = {'initial': ...}`
3. We can allow the experimental_initial_tablets tag to still be used to define a non-default initial_tablets. There should also be a way to disable tablets with a tag (maybe set to empty or none or 0 or something?) when creating a table (can't be changed afterwards...).
4. Recently test/cql-pytest/run.py was modified to enable tablets by default, so some Alternator tests (notably for Streams and TTL). We need to use in the affected Alternator tests a xfail_tablets tag like we added in test/cql-pytest.

**Comment by mykaul** [comment]
Moving to 6.1, as Alternator needs LWT, which is not supported in 6.0 with tablets. (https://github.com/scylladb/scylladb/issues/18068 )

**Comment by nyh** [comment]
What's the meaning of "moving" an issue that was already fixed?
If you're thinking of undoing this fix, you need to open a new issue asking to revert the fix.
And doing this would be very sad :-(

**Comment by mykaul** [comment]
> What's the meaning of "moving" an issue that was already fixed? If you're thinking of undoing this fix, you need to open a new issue asking to revert the fix. And doing this would be very sad :-(

Making sure it won't appear wrongly in the 6.0 release notes.

**Comment by nyh** [comment]
> Making sure it won't appear wrongly in the 6.0 release notes.

Was 6.0 branched already? If not, it already appears in the code!

**Comment by nyh** [comment]
To clarify (CC @bhalevy) what the patch for this issue (https://github.com/scylladb/scylladb/pull/16900) did was that if Scylla supports tablets, then every new Alternator table created by a `CreateTable` operation will use tablets **by default**. There's the possibility not to use tablets (by giving a tag), but the default is to use them.
Now, if suddenly LWT doesn't work with tablets, and Alternator often requires LWT, this default doesn't make sense. The default should be to create Alternator tables without tablets - so that Alternator will continue to work in 6.0, even if without tablets.

However, please don't blindly revert https://github.com/scylladb/scylladb/pull/16900 - this is an old patch that had a lot coming after it, and also a bunch of testing stuff. We need a new patch that changes the default and does minimal changes to the tests, if at all.

**Comment by nyh** [comment]
@bhalevy there shouldn't be almost any "churn to unit tests". Since all tests passed on Alternator without tablets, they will continue to all pass on Alternator without tablets (which will now become the default). Even tests which only work *with* tablets will be automatically skipped (because they won't be runnable if Alternator doesn't support tablets), so they don't need any changes.

If you and @mykaul already decided that this - namely, making tablets NOT the default on Alternator - needs to be done - please let me know and I can do it myself too. Until now I wasn't even sure what you guys decided to do. It will be sad for me to see Alternator not use the cool new tablets which everyone else will be using, but, what can I do :-( Hopefully somebody will work on LWT for 6.1 and we can return reenable tablets on Alternator too.
